kong/tools/public.lua

```diff
@@ -53,6 +53,10 @@ do
 
     [MIME_TYPES.json] = function()
       local raw_body  = ngx_req_get_body_data()
+      if not raw_body then
+        return {}, raw_body
+      end
```

Comment:
As per ngx_lua's docs: 

> This function returns nil if
>   the request body has not been read,
>   the request body has been read into disk temporary files,
>   or the request body has zero size.

As such, it would seem more appropriate to me if we were logging such occurrences, as they would probably highlight a programmer error or assumption (e.g. calling this method from a `GET` request, or not having wrongly tuned the `client_body_buffer_size`).

Maybe:
```
ngx_log(ERR, "could not read request body")
```

We could make it suggestive but I am afraid end-users could see it (and not only programmers) and it has three possible failure scenarios, which would make it cumber-stone... A simple error message should be good enough.